include/pybind11/stl_bind.h

```diff
@@ -657,9 +700,13 @@ template <typename Map, typename holder_type = std::unique_ptr<Map>, typename...
 class_<Map, holder_type> bind_map(handle scope, const std::string &name, Args &&...args) {
     using KeyType = typename Map::key_type;
     using MappedType = typename Map::mapped_type;
-    using KeysView = detail::keys_view<Map>;
-    using ValuesView = detail::values_view<Map>;
-    using ItemsView = detail::items_view<Map>;
+    using StrippedKeyType =
+        typename std::remove_cv<typename std::remove_reference<KeyType>::type>::type;
```

Comment:
detail::remove_cvref might be abetter fit here?

Reply:
Absolutely, but isn't it a C++-20 feature?

Reply:
@aimir We have a backport in detail/common.h

Reply:
Oh, whoops! I completely missed that you said `detail::` rather than `std::`.

Done in 41b8ffc893399815aa405ae67cc11d44e40ce22d.